homeassistant/components/tellduslive.py

```diff
@@ -17,141 +17,188 @@
         token = config[DOMAIN].get(CONF_TOKEN)
         token_secret = config[DOMAIN].get(CONF_TOKEN_SECRET)
 
         from tellive.client import LiveClient
-        from tellive.live import TelldusLive
 
-        self._sensors = []
-        self._switches = []
+        self._switches = {}
+        self._sensors = {}
+
+        self._hass = hass
+        self._config = config
 
         self._client = LiveClient(public_key=public_key,
                                   private_key=private_key,
                                   access_token=token,
                                   access_secret=token_secret)
-        self._api = TelldusLive(self._client)
-
-    def update(self, hass, config):
-        """ Send discovery event if component not yet discovered. """
-        self._update_sensors()
-        self._update_switches()
-        for component_name, found_devices, discovery_type in \
-            (('sensor', self._sensors, DISCOVER_SENSORS),
-             ('switch', self._switches, DISCOVER_SWITCHES)):
-            if len(found_devices):
-                component = get_component(component_name)
-                bootstrap.setup_component(hass, component.DOMAIN, config)
-                hass.bus.fire(EVENT_PLATFORM_DISCOVERED,
-                              {ATTR_SERVICE: discovery_type,
-                               ATTR_DISCOVERED: {}})
-
-    def _request(self, what, **params):
-        """ Sends a request to the Tellstick Live API. """
 
+    def validate_session(self):
+        """ Make a dummy request to see if the session is valid """
+        try:
+            return 'email' in self.request("user/profile")
+        except RuntimeError:
+            return False
+
+    def discover(self):
+        """ Update states, will trigger discover """
+        self.update_sensors()
+        self.update_switches()
+
+    def _discover(self, found_devices, component_name):
+        """ Send discovery event if component not yet discovered """
+
+        if not len(found_devices):
+            return
+
+        _LOGGER.info("discovered %d new %s devices",
+                     len(found_devices), component_name)
+
+        component = get_component(component_name)
+        bootstrap.setup_component(self._hass,
+                                  component.DOMAIN,
+                                  self._config)
+
+        discovery_type = DISCOVERY_TYPES[component_name]
+
+        self._hass.bus.fire(EVENT_PLATFORM_DISCOVERED,
+                            {ATTR_SERVICE: discovery_type,
+                             ATTR_DISCOVERED: found_devices})
+
+    def request(self, what, **params):
+        """ Sends a request to the tellstick live API """
         from tellive.live import const
 
         supported_methods = const.TELLSTICK_TURNON \
             | const.TELLSTICK_TURNOFF \
-            | const.TELLSTICK_TOGGLE
+            | const.TELLSTICK_TOGGLE \
+
+        #   | const.TELLSTICK_BELL \
```

Comment:
added note. these are unsupported stuff. just a reminder for the future. or for someone else with access to these kind of devices to implement.